lib: Pass absolute authfile path when pulling LBIs

csssuf opened the pull request:
ostree-ext explicitly handles authfile paths as relative; this works fine for most callers of get_global_authfile, as they only read the returned open file descriptor, and ignore the path. However, pulling logically bound images requires passing the actual authfile path to Podman, so we must resolve the absolute path in this case - otherwise, we see errors like the following:

```
[root@fedora ~]# bootc upgrade
layers already present: 69; layers needed: 1 (242.2 MB)
Fetched layers: 230.95 MiB in 3 seconds (90.88 MiB/s)
  Deploying: done (3 seconds)
  Fetching bound image: quay.io/prometheus/node-exporter:v1.10.2: done (0 seconds)
error: Upgrading: Staging: Pulling bound images: Pulling bound images: Failed to pull image: Subprocess failed: ExitStatus(unix_wait_status(32000))
Error: credential file is not accessible: faccessat etc/ostree/auth.json: no such file or directory
```

Since cap_std::fs::Dir intentionally does not expose its filesystem path, we must resort to reconstructing it from a file descriptor. We could do this by inspectingthe file descriptor for `sysroot` and combining that with the relative path returned by get_global_authfile, but since get_global_authfile returns the descriptor of the actual authfile, we can simply read that directly.

cgwalters reviewed (changes requested):
Thanks for finding this bug and trying a patch!

Can you explain if you built and tested this, or was this more of a drive-by?

csssuf commented:
I have built and tested it! From an earlier testing iteration with a debug print still in there:

```
[root@fedora ~]# bootc-local upgrade
Fetched layers: 0 B in 0 seconds (0 B/s)
  Deploying: done (3 seconds)
  Fetching bound image: quay.io/prometheus/node-exporter:v1.10.2
authfile_path: "/etc/ostree/auth.json"
  Fetching bound image: quay.io/prometheus/node-exporter:v1.10.2: done (1 second)
Bound images stored: 1
Pruned images: 0 (layers: 0, objsize: 726.8 MB)
Queued for next boot: 10.0.2.2:8000/localhost/test-bound-image
  Version: 43.20251213.0
  Digest: sha256:5cf3430f72d9dae0fc223f41ee13b5eec04e5d270369688a5b1415ce5047448b
Total new layers: 70    Size: 2.1 GB
Removed layers:   0     Size: 0 bytes
Added layers:     0     Size: 0 bytes
[root@fedora ~]#
```

cgwalters commented on crates/lib/src/podstorage.rs:
```diff
@@ -353,10 +353,15 @@ impl CStorage {
+        let authfile_fd =
+            ostree_ext::globals::get_global_authfile(&self.sysroot)?.map(|(_authfile, fd)| fd);
+        if let Some(fd) = authfile_fd {
+            let authfile_path = std::fs::read_link(format!("/proc/self/fd/{}", fd.as_raw_fd()))
```
I pushed a followup commit here, can you review it?
